common/domain/replication_queue_test.go

```diff
@@ -619,3 +710,28 @@ func TestPurgeAckedMessages(t *testing.T) {
 		})
 	}
 }
+
+func TestReplicationQueueImpl_purgeProcessor(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockQueue := persistence.NewMockQueueManager(ctrl)
+	rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil).(*replicationQueueImpl)
+	atomic.StoreInt32(&rq.status, common.DaemonStatusStarted)
+
+	done := make(chan bool)
+	mockQueue.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{}, nil).AnyTimes()
+	mockQueue.EXPECT().DeleteMessagesBefore(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	go func() {
+		rq.purgeProcessor()
+		close(done)
+	}()
+
+	time.Sleep(1 * time.Nanosecond)
+	rq.Stop()
+	select {
+	case <-done:
+		// Pass if the goroutine exits
+	case <-time.After(1 * time.Millisecond):
```

Comment:
1ms might be too aggressive of deadline. On CI machines with limited CPU resources this may cause flakyness. I recommend 10ms. If `done` is closed earlier we will not wait for 10ms anyway